API/SSW.Rewards/Services/NotificationHubsService.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.NotificationHubs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using SSW.Rewards.Application.Common.Interfaces;
+using SSW.Rewards.Application.Common.Models;
+
+namespace SSW.Rewards.WebAPI.Services
+{
+    public class NotificationHubService : INotificationService
+    {
+        readonly NotificationHubClient _hub;
+        readonly Dictionary<string, NotificationPlatform> _installationPlatform;
+        readonly ILogger<NotificationHubService> _logger;
+
+        public NotificationHubService(IOptions<NotificationHubOptions> options, ILogger<NotificationHubService> logger)
+        {
+            _logger = logger;
+            _hub = NotificationHubClient.CreateClientFromConnectionString(
+                options.Value.ConnectionString,
+                options.Value.Name);
+
+            _installationPlatform = new Dictionary<string, NotificationPlatform>
+            {
+                { nameof(NotificationPlatform.Apns).ToLower(), NotificationPlatform.Apns },
+                { nameof(NotificationPlatform.Fcm).ToLower(), NotificationPlatform.Fcm }
+            };
+        }
+
+        public async Task<bool> CreateOrUpdateInstallationAsync(DeviceInstall deviceInstallation, CancellationToken token)
+        {
+            if (string.IsNullOrWhiteSpace(deviceInstallation?.InstallationId) ||
+                string.IsNullOrWhiteSpace(deviceInstallation?.Platform) ||
+                string.IsNullOrWhiteSpace(deviceInstallation?.PushChannel))
+                return false;
+
+            var installation = new Installation()
+            {
+                InstallationId = deviceInstallation.InstallationId,
+                PushChannel = deviceInstallation.PushChannel,
+                Tags = deviceInstallation.Tags
+            };
+
+            if (_installationPlatform.TryGetValue(deviceInstallation.Platform, out var platform))
```

Comment:
Not really needed, but changing the function to something like the below makes it a bit nicer to read
i.e.
```csharp
try {

  if(!_installationPlatform.TryGetValue(deviceInstallation.Platform, out var platform)){
	return false
  }
  installation.Platform = platform;

  await _hub.CreateOrUpdateInstallationAsync(installation, token);
  return true;
} catch {}

return false
```

Reply:
All done - thanks, looks much better now!